tests/cell/cell_test_suite.cpp

```diff
@@ -462,6 +463,22 @@ class cell_test_suite : public test_suite
         xlnt_assert_equals(cell.number_format().format_string(), "dd--hh--mm");
     }
 
+    void test_number_format_id()
+    {
+        xlnt::workbook wb;
+        auto ws = wb.active_sheet();
+        auto cell1 = ws.cell("A1");
+        auto cell2 = ws.cell("A2");
+
+        xlnt::number_format format1("DD-MM-YYYY hh:mm:ss");
+        xlnt::number_format format2("DD-MM-YYYY hh:mm:ss");
+
+        cell1.number_format(format1);
+        cell2.number_format(format2);
+
+        xlnt_assert_equals(cell1.number_format().id(), cell2.number_format().id());
+    }
```

Comment:
Could you also add a negative test, i.e. check if a different `number_format` gets a different id?